src/main/java/com/redhat/devtools/lsp4ij/features/selectionRange/LSPExtendWordSelectionHandler.java

```diff
@@ -89,24 +88,11 @@ public List<TextRange> select(@NotNull PsiElement element,
             }
         }
 
-        // Get the selection ranges
-        List<SelectionRange> selectionRanges = LSPSelectionRangeSupport.getSelectionRanges(file, document, effectiveOffset);
-        if (ContainerUtil.isEmpty(selectionRanges)) {
-            return null;
-        }
-
-        // Convert the selection ranges into text ranges
-        Set<TextRange> textRanges = new LinkedHashSet<>(selectionRanges.size());
-        for (SelectionRange selectionRange : selectionRanges) {
-            TextRange selectionTextRange = LSPSelectionRangeSupport.getTextRange(selectionRange, document);
-            textRanges.addAll(expandToWholeLinesWithBlanks(editorText, selectionTextRange));
-
-            for (SelectionRange parentSelectionRange = selectionRange.getParent();
-                 parentSelectionRange != null;
-                 parentSelectionRange = parentSelectionRange.getParent()) {
-                TextRange parentSelectionTextRange = LSPSelectionRangeSupport.getTextRange(parentSelectionRange, document);
-                textRanges.addAll(expandToWholeLinesWithBlanks(editorText, parentSelectionTextRange));
-            }
+        // Get the selection ranges and extend them to whole lines
+        Set<TextRange> textRanges = new LinkedHashSet<>();
+        List<TextRange> selectionTextRanges = LSPSelectionRangeSupport.getSelectionTextRanges(file, editor, effectiveOffset);
```

Comment:
Again, this is the result of encapsulation of selection range-to-text range conversion.